client/src/modules/lead/AddLead/AddLead.js

```diff
@@ -265,10 +271,14 @@ class AddLead extends React.Component {
     this.setState({
       ...this.state,
       name: this.getNameValue(),
+      contact: {
+        id: this.state.contactAfterSelect.id,
+        name: trim(this.state.contact.name),
+      },
       namePlaceholder: this.getPlaceholderValue(),
       nameChanged: isBlank(this.state.organization.name) && isBlank(this.state.contact.name) ? false : this.state.nameChanged,
       openContactDropdown: false,
-      showContactBadge: this.state.contact.name.length > 0 && this.state.afterContactSelectShowBadge && !this.state.contact.id,
+      showContactBadge: this.state.contact.name.length > 0 && this.state.afterContactSelectShowBadge && !this.state.contact.id && !isBlank(this.state.contact.name),
```

Comment:
2 same condition this.state.contact.name > 0 && !isBlank(this.state.contact.name)